x-pack/plugin/esql/qa/server/heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/heap_attack/HeapAttackIT.java

```diff
@@ -57,7 +57,6 @@ public class HeapAttackIT extends ESRestTestCase {
     @ClassRule
     public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
         .distribution(DistributionType.DEFAULT)
-        .nodes(1)
```

Comment:
Setting node count isn't supported in Serverless.
Considering that `1` is the default, I think the easiest is to just remove the setting. This would be preferred IMO to "detecting" the environment.